tests/devices/unit_tests/test_undulator_dcm.py

```diff
@@ -60,80 +66,90 @@ def test_correct_closest_distance_to_energy_from_table(dcm_energy, expected_outp
-    fake_undulator_dcm.undulator.current_gap.sim_put(5.3)  # type: ignore
-    fake_undulator_dcm.undulator.gap_motor.move = MagicMock()
-    fake_undulator_dcm.dcm.energy_in_kev.move = MagicMock()
+    set_sim_value(fake_undulator_dcm.undulator.gap_access, UndulatorGapAccess.DISABLED)
+    set_sim_value(fake_undulator_dcm.undulator.current_gap, 5.3)
+    set_sim_value(fake_undulator_dcm.dcm.energy_in_kev.readback, 5.7)
+
+    set_sim_value(fake_undulator_dcm.undulator.gap_motor.setpoint, 0.0)
+    set_sim_value(fake_undulator_dcm.undulator.gap_motor.readback, 0.0)
+
+    # fake_undulator_dcm.undulator.gap_motor.move = MagicMock()
+    # fake_undulator_dcm.dcm.energy_in_kev.move = MagicMock()
     mock_load.return_value = np.array([[5700, 5.4606], [7000, 6.045], [9700, 6.404]])
-    fake_undulator_dcm.dcm.energy_in_kev.user_readback.sim_put(5.7)  # type: ignore
 
-    fake_undulator_dcm.energy_kev.set(6.9)
+    await fake_undulator_dcm.set(6.9)
+
+    assert (await fake_undulator_dcm.dcm.energy_in_kev.setpoint.get_value()) == 6.9
+    # Verify undulator has not been asked to move
+    assert (await fake_undulator_dcm.undulator.gap_motor.setpoint.get_value()) == 0.0
 
-    fake_undulator_dcm.dcm.energy_in_kev.move.assert_called_once_with(6.9, timeout=30)
-    fake_undulator_dcm.undulator.gap_motor.move.assert_not_called()
     mock_logger.info.assert_called()
+    mock_logger.debug.assert_called_once()
 
 
 @patch("dodal.devices.undulator_dcm.loadtxt")
 @patch("dodal.devices.undulator_dcm.LOGGER")
-def test_if_gap_is_already_correct_then_dont_move_gap(
+async def test_if_gap_is_already_correct_then_dont_move_gap(
     mock_logger: MagicMock, mock_load: MagicMock, fake_undulator_dcm: UndulatorDCM
 ):
-    fake_undulator_dcm.undulator.gap_motor.move = MagicMock()
-    fake_undulator_dcm.dcm.energy_in_kev.move = MagicMock()
+    set_sim_value(fake_undulator_dcm.dcm.energy_in_kev.setpoint, 0.0)
+    set_sim_value(fake_undulator_dcm.dcm.energy_in_kev.readback, 0.0)
+    # fake_undulator_dcm.undulator.gap_motor.move = MagicMock()
+    # fake_undulator_dcm.dcm.energy_in_kev.move = MagicMock()
     mock_load.return_value = np.array([[5700, 5.4606], [7000, 6.045], [9700, 6.404]])
-    fake_undulator_dcm.undulator.current_gap.sim_put(5.4605)  # type: ignore
+    set_sim_value(fake_undulator_dcm.undulator.current_gap, 5.4605)
 
-    fake_undulator_dcm.energy_kev.set(5.8).wait(timeout=0.01)
+    status = fake_undulator_dcm.set(5.8)
+    await asyncio.wait_for(status, timeout=0.01)
 
-    fake_undulator_dcm.undulator.gap_motor.move.assert_not_called()
+    # Verify undulator has not been asked to move
+    assert (await fake_undulator_dcm.undulator.gap_motor.setpoint.get_value()) == 0.0
     mock_logger.info.assert_called_once()
+    mock_logger.debug.assert_called_once()
 
 
-def test_energy_set_only_complete_when_all_statuses_are_finished(
+async def test_energy_set_only_complete_when_all_statuses_are_finished(
```

Comment:
Thanks, this is a nice pattern for testing this that I hadn't yet worked out for `ophyd_async`!